test/net/sourceforge/kolmafia/session/ResultProcessorTest.java

```diff
@@ -252,4 +257,54 @@ public void gettingSecondMaelstromStone(int stone) {
       }
     }
   }
+
+  @Nested
+  class Cookbookbat {
+    private static Stream<AdventureResult> coobookbatRecipes() {
+      return Stream.of(
+          ItemPool.get(ItemPool.ROBY_BORIS_BEER),
+          ItemPool.get(ItemPool.ROBY_HONEY_BUN_OF_BORIS),
+          ItemPool.get(ItemPool.ROBY_RATATOUILLE_DE_JARLSBERG),
+          ItemPool.get(ItemPool.ROBY_JARLSBERGS_VEGETABLE_SOUP),
+          ItemPool.get(ItemPool.ROBY_PETES_WILY_WHEY_BAR),
+          ItemPool.get(ItemPool.ROBY_PETES_SNEAKY_SMOOTHIE),
+          ItemPool.get(ItemPool.ROBY_BORIS_BREAD),
+          ItemPool.get(ItemPool.ROBY_ROASTED_VEGETABLE_OF_J),
+          ItemPool.get(ItemPool.ROBY_PETES_RICH_RICOTTA),
+          ItemPool.get(ItemPool.ROBY_ROASTED_VEGETABLE_FOCACCIA),
+          ItemPool.get(ItemPool.ROBY_PLAIN_CALZONE),
+          ItemPool.get(ItemPool.ROBY_BAKED_VEGGIE_RICOTTA),
+          ItemPool.get(ItemPool.ROBY_DEEP_DISH_OF_LEGEND),
+          ItemPool.get(ItemPool.ROBY_CALZONE_OF_LEGEND),
+          ItemPool.get(ItemPool.ROBY_PIZZA_OF_LEGEND));
+    }
+
+    @ParameterizedTest
+    @MethodSource("coobookbatRecipes")
+    public void cookbookbatPropertyGetsUpdated(AdventureResult recipe) {
+      var cleanups =
+          new Cleanups(
+              withFamiliar(FamiliarPool.COOKBOOKBAT), withProperty("_cookbookbatRecipeDrops", ""));
+
+      try (cleanups) {
+        ResultProcessor.processResult(true, recipe);
+
+        assertThat("_cookbookbatRecipeDrops", isSetTo("1"));
+      }
+    }
+
+    @Test
+    public void cookbookbatPropertyNoUpdateIfNotActiveFamiliar() {
```

Comment:
Change this test to not check active familiar - since that's irrelevant - to checking with combatResults false.
```
ResultProcessor.processResult(false, ItemPool.get(ItemPool.ROBY_BAKED_VEGGIE_RICOTTA))
```